apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -594,21 +622,24 @@ def flash(self, options):
             )
 
     def open_transport(self, options):
+        # TODO: This is to avoid breaking GPU docker on running the tutorials.
         import serial
         import serial.tools.list_ports
```

Comment:
Can you clarify this comment a bit? I don't understand how this fixes the GPU docker on tutorials.

Reply:
When I moved these imports to the top of the file, there was an error with doc stage in GPU. I think the reason is that we don't have dependencies for this in GPU docker. cc @driazati

Reply:
if this file is imported but this function isn't run during the tutorial docs build then it would fail, this fixes it with a lazy import. it would be nice to link to an issue about this though for more context

Reply:
@driazati I remember there was some discussion about running tutorials in the related docker and only render the results in GPU/Doc stage. Did that happen?

Reply:
https://github.com/apache/tvm/issues/13540

Reply:
No we never implemented anything like that, though here it might be easiest to just install `pyserial` in the GPU docker image and call it a day